### PR TITLE
Changing the 404 error messaging to not include the "joke" as well as…

### PR DESCRIPTION
**Pull request by nateynateynate**
… provides a link to the homepage.

### Description
Addresses issue #1181 
 
### Issues Resolved
Removes the 'ironic' joke as well as provides a link to the homepage. 

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: 404.markdown [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 ---
 
-Perhaps you mistyped the URL or we've moved something around. Sorry 'bout that.
\ No newline at end of file
+Sorry, that page doesn't exist. Perhaps you'd like to visit our [homepage](https://www.opensearch.org).
+Or, try one of these popular pages.
+
+
+* Read the [blog](https://www.opensearch.org/blog)
+* Find out about upcoming [events](https://www.opensearch.org/events/calendar/2024-04.html)
```
Comment (krisfreedain) [inline 404.markdown:14]: should link to https://opensearch.org/events/

File: 404.markdown [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+* Join the conversation on the [forum](https://forum.opensearch.org/)
+* Join [slack](https://opensearch.org/slack.html)
```
Comment (krisfreedain) [inline 404.markdown:16]: [slack] --- should be [Slack]

**Review by krisfreedain — CHANGES_REQUESTED** [review]
@nateynateynate a couple edits needed

File: 404.markdown (hunk 2 above)
Comment (nateynateynate) [reply 404.markdown]: ok all set.

File: 404.markdown (hunk 1 above)
Comment (nateynateynate) [reply 404.markdown]: All set.

**Review by krisfreedain — APPROVED** [review]
Woot!

**Pull request merged**